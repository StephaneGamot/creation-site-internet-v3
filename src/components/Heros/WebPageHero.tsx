"use client";

import { motion } from "framer-motion";
import LetsTalkButton from "@/components/Buttons/LetsTalkButton";
import { useTranslations } from "next-intl";

export default function WebPageHero(){
  const t = useTranslations("webHero");

  return (
    <section className="text-center pt-12 max-w-4xl mx-auto mb-12">
      <motion.h1
        initial={{ opacity: 0, y: 40 }}
        animate={{ opacity: 1, y: 0 }}
        transition={{ duration: 0.8 }}
        className="text-4xl md:text-6xl font-title font-semibold mb-6"
      >
        {t("title")}
      </motion.h1>
      <p className="text-lg md:text-xl text-gray-700 mb-8">
        {t("text")}
      </p>
      <LetsTalkButton />
    </section>
  );
}
