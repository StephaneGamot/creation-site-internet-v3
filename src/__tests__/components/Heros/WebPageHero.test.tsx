import React from "react";
import { render, screen } from "@testing-library/react";
import WebPageHero from "@/components/Heros/WebPageHero";
import { axe } from "jest-axe";
import renderer from "react-test-renderer";

// ✅ Mock de next-intl
jest.mock("next-intl", () => ({
  useTranslations: () => (key: string) => {
    const dict: Record<string, string> = {
      title: "Créons un site web qui vous ressemble",
      text: "Des sites performants, élégants et orientés conversion.",
    };
    return dict[key] || "";
  },
}));

// ✅ Mock de framer-motion
jest.mock("framer-motion", () => ({
  motion: {
    h1: (props: any) => <h1 {...props} />,
  },
}));

// ✅ Mock du bouton LetsTalkButton
jest.mock("@/components/Buttons/LetsTalkButton", () => () => (
  <button>Parlons-en</button>
));

describe("WebPageHero", () => {
  it("affiche le titre, le texte et le bouton", () => {
    render(<WebPageHero />);
    expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent("Créons un site web qui vous ressemble");
    expect(screen.getByText("Des sites performants, élégants et orientés conversion.")).toBeInTheDocument();
    expect(screen.getByRole("button")).toHaveTextContent("Parlons-en");
  });

  it("passe les tests d’accessibilité", async () => {
    const { container } = render(<WebPageHero />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it("correspond au snapshot", () => {
    const tree = renderer.create(<WebPageHero />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
