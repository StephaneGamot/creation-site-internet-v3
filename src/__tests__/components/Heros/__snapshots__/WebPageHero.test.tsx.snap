// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`WebPageHero correspond au snapshot 1`] = `null`;
